src/microbe_stage/Patch.cs

```diff
@@ -79,6 +79,35 @@ public bool AddNeighbour(Patch patch)
         return Adjacent.Add(patch);
     }
 
+    /// <summary>
+    ///   Checks all neighbors recursively to find all connected patch nodes
+    /// </summary>
+    /// <returns>A <see cref="HashSet{T}"/> of <see cref="Patch"/> connected to this node by some means</returns>
+    public HashSet<Patch> GetAllConnectedPatches()
+    {
+        HashSet<Patch> resultSet = new HashSet<Patch>();
+        CollectNeighbors(this, ref resultSet);
+
+        return resultSet;
+    }
+
+    /// <summary>
+    ///   Adds all neighbors recursively to the referenced <see cref="HashSet{T}"/>
+    /// </summary>
+    /// <param name="patch">The <see cref="Patch"/> to start from</param>
+    /// <param name="set">The <see cref="HashSet{T}"/> to add to</param>
+    public void CollectNeighbors(Patch patch, ref HashSet<Patch> set)
```

Comment:
```suggestion
    public void CollectNeighbours(Patch patch, HashSet<Patch> neighbours)
```

Besides that suggestion: I have a question, why does this take in a patch? This is a non-static method, why not use `this.Adjacent`. Why is it like this?

Reply:
Because this method calls itself on line 106 with the neighbour passed in as what to collect from. I probably should have made it static?

Reply:
Yes, that would make sense.

Reply:
I've converted it to static and moved it above all non static members. Along with all the other requested changes. It should look good now.